### ISSUE TITLE
Feature request: Cogo calculation

### ISSUE THREAD

**Issue by mittakainen** [opengisch/QField#1846]
It would be nice to have simple cogo calculation option when the needed point is is not accesible. I think the most versatile cogo option would be the calculation from baseline with distance and off-set. Baseline could be defined by already collected points or temporary points.

Example for baseline cogo calculation is shown e.g im trimble access help portal: [calculate point from baseline](https://help.trimblegeospatial.com/TrimbleAccess/2019.00/en/Cogo_Compute_Point.htm?tocpath=Key%20in%20%26%C2%A0Cogo%7CCogo%20calculations%7C_____2)

**Comment by m-kuhn** [comment]
Nice idea!

**Comment by ntrip6** [comment]
This has a lot of potential it opens up comm from a laser range finder for example
https://www.lasertech.com/LaserSoft-Measure-App.aspx
Laser range finders are useful for mapping into areas where GNSS does not function like dense forest, under overhangs, and in buildings.

**Comment by Sickculture** [comment]
yeah, this would be nice to set a point with distance and bearing. Also a plugin for rangefinders like BOSCH GLM series.

**Comment by nirvn** [comment]
@mittakainen , @ntrip6 , @Sickculture , this is becoming a reality, there's a PR for you to test out right here: https://github.com/opengisch/QField/pull/6923 :)

**Comment by ntrip6** [comment]
Fantastic! Thanks. I tested it briefly on a windows desktop.
How hard would it be to add support for Z elevation? Z is presumably needed for receiving relative (distance/direction) from a 3D laser range finder, and for creating regular field sampling points across a slope. We'd like to use slope distance instead of isotropic X,Y distances in the sampling grid.

**Comment by nirvn** [comment]
@ntrip6 , can you give me an actual example of how you'd use the reading into parameters here?

**Comment by nirvn** [comment]
@ntrip6 , I think I understand what you mentioned here. In the last builds of the PR, the point parameter has gained Z dimension awareness (provided the vector layer being digitized against has a geometry type containing the Z dimension), and an elevation parameter will be added when doing distance/angle from point:

<img width="494" height="865" alt="Image" src="https://github.com/user-attachments/assets/465ab7df-c7b9-4751-8e47-86fc15b239bf" />

**Comment by ntrip6** [comment]
Terrific I'm looking forward to testing it out.
We either use RTK GNSS derived Z values or query a high res underlying layer (such as lidar derived raster) for the Z and generate slope distance from that.
In terms of explaining applications I put out a query about this on the QGIS reddit thread and there's a discussion (and code) for 3d grid surface using slope distance using the Geometry Generator
https://www.reddit.com/r/QGIS/comments/1m42pxf/comment/n42rzac/
This improvement to QField is potentially even better because it can be calculated for each row of a grid instead of assuming the slope is consistent across the grid.

**Comment by m-kuhn** [comment]
I am very much looking forward to see this QField feature put into action and I'd be amazed to see anyone going for a followup and integrate laser rangefinders 💚

Unrelated: Amazing what's done with geometry generators nowadays. I didn't expect this in my wildest dreams when implementing this.
